environs/interface.go

```diff
@@ -60,6 +61,24 @@ type EnvironProvider interface {
 	PrepareConfig(context.Context, PrepareConfigParams) (*config.Config, error)
 }
 
+// ModelConfigProvider represents an interface that a [EnvironProvider] can
+// implement to provide opinions and defaults into a models config.
+type ModelConfigProvider interface {
```

Comment:
It is the intent to replace this long term. It is used in lots of places and I don't think we are ready to pull it. I could try and do it in this PR but it will be a lot of noise.